build_tools/docker/manage_prod.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Uses prod_digests.txt to update GCR's :prod tags.
+
+Usage:
+  Pull all images that should have :prod tags, tag the with :prod and push
+  them to GCR. This will make sure that you are at upstream head on the main
+  branch before pushing:
+    python3 build_tools/docker/manage_prod.py
+
+  Pull all images that should have :prod tags:
+    python3  build_tools/docker/manage_prod.py --pull_only
+"""
+
+import argparse
+import utils
+
+
+def parse_arguments():
+  """Parses command-line options."""
+  parser = argparse.ArgumentParser(
+      description="Pull and push the images in prod_digests.txt to GCR.")
+  parser.add_argument("--pull_only",
+                      "--pull-only",
+                      action="store_true",
+                      help="Pull but do not tag or push the images.")
+  return parser.parse_args()
+
+
+if __name__ == "__main__":
+  args = parse_arguments()
+
+  if not args.pull_only:
+    # Ensure the user has the correct authorization if they try to push to GCR.
+    utils.check_gcloud_auth()
+
+    # Only allow the :prod tag to be pushed from the version of
+    # `prod_digests.txt` at upstream HEAD on the main branch.
+    utils.run_command(["git", "sync"])
```

Comment:
This is a custom alias. We shouldn't use it in scripts. You can use https://github.com/google/iree/blob/main/scripts/git/git_update.sh if this is really necessary although I'm not sure how well environment variables propagate between python and bash and back again.

Reply:
Updated to use `git_update.sh`. Was the concern that some people might have a custom value set to `UPSTREAM_REMOTE` that isn't `upstream`?

Reply:
Yeah and I'm not sure if that threads through bash -> python -> bash

Reply:
Modified `git_update.sh` to echo the variable to check:

```shell
export UPSTREAM_REMOTE="etomer_maertspu"
python3
>>> import subprocess
>>> subprocess.run(["scripts/git/git_update.sh", "main"])
etomer_maertspu
```